### PR TITLE
Introduce gulp serve.

### PR DESCRIPTION
**Pull request by Carreau**
Watch the files, and insert a middleware that will rebuild, and reload
the page if a change is detected to the source.

--- 

Still not super familiar with gulp though, but works here on examples.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -145,11 +150,35 @@ gulp.task('dist', function() {
...
+        gulp.src('./')
+            .pipe(webserver({
+                root: 'examples',
+                livereload: true,
+                directoryListing: true,
+                fallback: 'index.html',
+                enable: true,
+                open: true
+        }))
+
```
Comment (sccolbert) [inline gulpfile.js]: I use 2 spaces for indentation for JS and TS

File: gulpfile.js [hunk 2]
```diff
@@ -167,13 +196,15 @@ gulp.task('examples', function() {
...
+    .pipe(gulp.dest('examples'))
+    .pipe(livereload());
 
   var css = gulp.src('examples/**/index.styl')
     .pipe(stylus({use: [nib()]}))
     .pipe(rename(function (path) {
       path.dirname += '/build'; }))
-    .pipe(gulp.dest('examples'));
+    .pipe(gulp.dest('examples'))
+    .pipe(livereload());
```
Comment (sccolbert) [inline gulpfile.js]: what does livereload() do here?

File: gulpfile.js (hunk 1 above)
Comment (Carreau) [inline gulpfile.js]: Ah, sorry, will change

File: gulpfile.js (hunk 2 above)
Comment (Carreau) [inline gulpfile.js]: As far as I understand, you call it to trigger the livereloading after build.

**Pull request closed without merging**